### PR TITLE
Csv ticketing

### PR DESCRIPTION
**Pull request by b-destoop**
Added a modification to the admin of online orders so that the curator of the orchestra can copy-paste names out of excel into a text field to bulk-update them and set them to paid. 

(Matters a lot for 1500 sold tickets)

### CODE REVIEW

File: admin.py [hunk 1]
```diff
@@ -107,6 +109,38 @@ class TicketInline(admin.TabularInline):
     extra = 0
 
 
+def process_names_csv(csv_file):
+    """
+    Process a csv file for online orders
+    @param csv_file: A csv file containing names to be set to paid.
+    @return: list of names that had problems being processed and a message summarizing the done process.
+    """
+    names = pandas.read_csv(csv_file, delimiter="\t", header=None)
```
Comment (WimVanRoy) [inline admin.py]: At this moment, the CSV should be formatted as "First name" , "Last name". Could we also make a column "Order key"?

File: admin.py [hunk 2]
```diff
@@ -107,6 +109,38 @@ class TicketInline(admin.TabularInline):
...
+    open_tickets = OnlineOrder.objects.filter(performance__production__active=True).all()
```
Comment (WimVanRoy) [inline admin.py]: To avoid future confusion, rename to open_orders

File: admin.py [hunk 3]
```diff
@@ -107,6 +109,38 @@ class TicketInline(admin.TabularInline):
...
+    amount_processed_fail = 0
+    for index, data in names.iterrows():
+        first_name = data[0]
+        last_name = data[1]
+        person_tickets = open_tickets.filter(first_name__iexact=first_name, last_name__iexact=last_name)
+        if not person_tickets.exists():
+            error_msgs.append(f"'{first_name} {last_name}' was not found to have tickets in active performances.")
+            amount_processed_fail += 1
+            continue
+        person_tickets = person_tickets.all()
```
Comment (WimVanRoy) [inline admin.py]: Woops, this is not how we set a ticket to payed and send out the mails. You should use:

views.send_order_payed(request, id)

Be aware you have to pass request to be able to send out a mail...

File: admin.py [hunk 4]
```diff
@@ -107,6 +109,38 @@ class TicketInline(admin.TabularInline):
...
+
+        for ticket in person_tickets:
```
Comment (WimVanRoy) [inline admin.py]: I don't think this is a clever idea since I notice a lot of double orders by accident!

File: admin.py (hunk 1 above)
Comment (b-destoop) [reply admin.py]: I think that negates the usefulness for the orchestra administrator. If the administrator needs to keep track of the order keys, they needs to look up these keys in the website database, which is slow and one of the main reasons that this feature needs to exist in the first place.

**Comment by WimVanRoy** [comment]
Not necessarily since the order ID is part of the payment message. So if you get your data from your bank, it is in there. (Or should be)

I would offer it as an extra option. Either you provide the payment message or you don't. Furthermore, I would suggest to also add a payment price. Since we are handling real payments here from People outside the orchestra, you can not allow having tickets validated which are not payed yet.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Pull request merged**